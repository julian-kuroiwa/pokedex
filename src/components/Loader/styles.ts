import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100%{
    transform: rotate(360deg);
  }
`;

export const Container = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 20px 0 40px;
`;

export const Pokeball = styled.div`
  width: 60px;
  height: 60px;
  background-color: #fff;
  border-radius: 50%;
  position: relative;
  overflow: hidden;
  border: 3px solid;
  animation: ${rotate} 0.8s linear 0s infinite;

  &::after {
    content: '';
    position: absolute;
    width: 60px;
    height: 30px;
    background-color: red;
    border-bottom: 4px solid;
    top: -4px;
  }

  &::before {
    content: '';
    position: absolute;
    background-color: #fff;
    width: 10px;
    height: 10px;
    border: 4px solid;
    border-radius: 50%;
    bottom: 18px;
    right: 18px;
    z-index: 1;
  }
`;
