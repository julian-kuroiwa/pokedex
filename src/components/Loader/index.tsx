import React from 'react';

import { Container, Pokeball } from './styles';

const Loader: React.FC = () => (
  <Container>
    <Pokeball />
  </Container>
);

export default Loader;
