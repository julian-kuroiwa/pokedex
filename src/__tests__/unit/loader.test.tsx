import { shallow } from 'enzyme';
import React from 'react';
import Loader from '../../components/Loader';

describe('Loader', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Loader />);

    expect(wrapper).toMatchSnapshot();
  });
});
