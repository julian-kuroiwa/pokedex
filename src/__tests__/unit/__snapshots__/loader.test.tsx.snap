// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match snapshot 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;
